frontend: show passphrase configuration in fullscreen

thisconnect opened the pull request:
When enabling or disabling the passphrase setting users should
fully focus on the information shown and not be distracted by other
elements on the page, therefore the view in fullscreen is more
suitable than just a dialog.

Move call to replug-the-device to be the last message of the
success view.

Introduced a new `<Main>` component for contents of the page, this
component should replace `<div class="container">`. Also left example
code of how `<Header>` and `<Guide>` components can be used.

Made also sure the `<View>` content is scrollable inside `<Main>`.

Sometimes the BitBoxApp requires the user to focus on the device
and confirm or reject something on the BitBox02.

Turned caret pointing to bb02 device into a shared component, so
it can be re-used in different places like AOPP.

thisconnect committed:
frontend: show passphrase configuration in fullscreen

When enabling or disabling the passphrase setting users should
fully focus on the information shown and not be distracted by other
elements on the page, therefore the view in fullscreen is more
suitable than just a dialog.

Move call to replug-the-device to be the last message of the
success view.

Introduced a new <Main> component for contents of the page, this
component should replace <div class="container">. Also left example
code of how <Header> and <Guide> components can be used.

Made also sure the <View> content is scrollable inside <Main>.

Sometimes the BitBoxApp requires the user to focus on the device
and confirm or reject something on the BitBox02.

Turned caret pointing to bb02 device into a shared component, so
it can be re-used in different places like AOPP.

x1ddos commented on frontends/web/src/locales/en/app.json:
```diff
@@ -1027,6 +1027,7 @@
     },
     "successDisabled": {
       "message": "Optional passphrase <strong>successfully enabled</strong>!\nYou’ll be asked to provide a passphrase from now on.",
+      "messageEnd": "Please replug your BitBox02 now.",
```
it could be someone else's bitbox02 :)
how about this: Please replug the BitBox02 device now.

thisconnect replied:
I agree, directly addressing the user can have a strong effect, i.e. when we really want to make them aware about something the should or shouldn't do.

But the effect is kind of gone if we always use direct addressing.

I will go through the text and propose some changes.

x1ddos commented on frontends/web/src/routes/device/bitbox02/passphrase.tsx:
```diff
@@ -251,11 +280,15 @@ class Passphrase extends Component<Props, State> {
+                )}
+                {/* <Guide>
+                    <Entry key="whatisapassphrase" entry={t('guide.passphrase.whatisapassphrase')} />
+                </Guide> */}
```
is the guide entry not ready yet?

thisconnect replied:
Currently there is no additional guide for passphrase, but we now make the user read through the information instead of providing an optional guide.

Is there something missing that you'd like to see in the guide?

If yes then depending on what kind of information it is we should consider adding it to the information shown. Maybe the guide could contain some technical details that we don't want to show to regular users?

x1ddos reviewed (commented):
very nice! i love it how there's no more distractions, just the main content.

there's one inconsistency. whem i'm on any page like this one:

![image](https://user-images.githubusercontent.com/25405/142406465-d54d3d45-6b52-4cf8-bbed-9c2ec5848778.png)

and before ticking the "I understand ..." checkbox i realize i don't understand and want to go back. i click the "back" button, expecting it to go the the previous screen but it closes the whole thing.

thisconnect commented:
Thanks for testing,

I changed
- the back button to actually go 1 step back instead of going back to the setting view
- added a close button to go back to the settings view
- updated the styling especially of summary page

For now I opened a new PR here https://github.com/digitalbitbox/bitbox-wallet-app/pull/1478 but this is only temporary I'll check Monday if and how to split them up into different commits.

![Screen Shot 2021-11-20 at 12 08 58 PM](https://user-images.githubusercontent.com/546900/142724705-a6e4871a-b7b2-4d52-a9fd-c8e88d74bc84.png)

![Screen Shot 2021-11-20 at 12 09 07 PM](https://user-images.githubusercontent.com/546900/142724707-d4e20f3c-bbd7-4368-9352-f664bef7e366.png)

thisconnect commented:
superseded by https://github.com/digitalbitbox/bitbox-wallet-app/pull/1478

x1ddos commented:
> superseded by #1478

so, close this?
